CMakeLists.txt

```diff
@@ -125,7 +125,7 @@ ENDIF()
 
 set(CMAKE_C_EXTENSIONS OFF)
 set(CMAKE_CXX_EXTENSIONS OFF)
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_C_STANDARD 11)
```

Comment:
Would it make sense to bump this to C17 as well? [C17 is a bugfix release](https://stackoverflow.com/questions/47529854/what-is-c17-and-what-changes-have-been-made-to-the-language). For new development, I'd argue it'd make sense to rely these bugfixes.

Reply:
I only checked whether the proposed compilers offered C++17 support, not C17; a cursory glance at Wikipedia tells me that, yes, even with maint-3.9 compilers we can do that, probably. I really don't expect it to *change* anything, though.

Reply:
Re. C17, https://gitlab.kitware.com/cmake/cmake/-/issues/17755  I am not sure why a fix has not been made in cmake. (I only cross checked the doc)